[Feature] Introduce Woo Watch App

Ecarrion opened the pull request:
closes #12690 

## Feature branch for the Watch App

- https://github.com/woocommerce/woocommerce-ios/pull/12647
- https://github.com/woocommerce/woocommerce-ios/pull/12663
- https://github.com/woocommerce/woocommerce-ios/pull/12683
- https://github.com/woocommerce/woocommerce-ios/pull/12709
- https://github.com/woocommerce/woocommerce-ios/pull/12751 
- https://github.com/woocommerce/woocommerce-ios/pull/12771
- https://github.com/woocommerce/woocommerce-ios/pull/12774
- https://github.com/woocommerce/woocommerce-ios/pull/12778
- https://github.com/woocommerce/woocommerce-ios/pull/12798
- https://github.com/woocommerce/woocommerce-ios/pull/12807
- https://github.com/woocommerce/woocommerce-ios/pull/12817
- https://github.com/woocommerce/woocommerce-ios/pull/12825
- https://github.com/woocommerce/woocommerce-ios/pull/12836
- https://github.com/woocommerce/woocommerce-ios/pull/12856
- https://github.com/woocommerce/woocommerce-ios/pull/12898

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

Ecarrion committed:
Adds initial WooWatchApp setup

Ecarrion committed:
Disable automatic sign in for alpha and release phases

iangmaia committed:
Configure Fastlane signing automation for the Watch app

iangmaia committed:
Configure project to use the new watch Enterprise and AppStore profiles

Ecarrion committed:
Update alpha companion app bundle id

Ecarrion committed:
Add temporary raw type in credential. May need to remove the extension in SessionManager later

Ecarrion committed:
First draft of dependency synchronizer from phone to watch

Ecarrion committed:
First draft of the phone depenency synchronizer

Ecarrion committed:
Copy credentials file

Ecarrion committed:
First draft of storing and fetching dependencies on the watch

Ecarrion committed:
Update code to support both platforms (watch and phone)

Ecarrion committed:
Update Podfile to add dependencies to the NetworkingWatchOS target

Ecarrion committed:
Configure project settings

Ecarrion committed:
Use new NetworkingWatchOS module

Ecarrion committed:
Exclude arm64 arch from Woo Watch target

Ecarrion committed:
Add dependencies to what app target

Ecarrion committed:
Update project settings to include
- Networking dependencies
- User Default dependencies
- Specific watch app compilation flags

Ecarrion committed:
Copy some files from WordPressShared to not have to copy the entire dependency

Ecarrion committed:
Add missing imports from Networking so watch app target can compile correctly.

Ecarrion committed:
Resolve conflicts with temporary workaround

Ecarrion committed:
Add Documentation and extract Dependencies serialization

Ecarrion committed:
Move credential type to credential model

Ecarrion committed:
Extract watch dependencies to it's own object together with their serialization

Ecarrion committed:
Polish syncronizers

Ecarrion committed:
Cleanup project

Ecarrion committed:
Fix swiftlint

Ecarrion committed:
Remove uneeded settings

Ecarrion committed:
Remove temporary WooFoundation watch0s

Ecarrion committed:
Add files to properly fire network requests

Ecarrion committed:
Add compiler conditionals for NetworkingWatchOS.Framework

Ecarrion committed:
Add keychain dependency

Ecarrion committed:
Remove unnecessary code

Ecarrion committed:
Remove duplicated reference to NetworkingWatchOS.framework

Ecarrion committed:
Placeholder commit

Ecarrion committed:
Update imports due to the new framework

Ecarrion committed:
[Watch App] Adds initial WooWatchApp setup (#12647)

Ecarrion committed:
[Watch] Add Networking framework to Watch Target (#12663)

dangermattic commented:
<!--
  0 Errors
  3 Warnings: View files have been modified,...
  1 Message: <p>This PR contains changes to...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          3 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">View files have been modified, but no screenshot or video is included in the pull request. Consider adding some for clarity.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 500 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is not assigned to a milestone.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false"><p>This PR contains changes to Tracks-related logic. Please ensure (<strong>author and reviewer</strong>) the following are completed:</p>

<ul>
  <li>The tracks events must be validated in the Tracks system.</li>
  <li>Verify the internal Tracks spreadsheet has also been updated.</li>
  <li>Please consider registering any new events.</li>
  <li>The PR must be assigned the <strong>category: tracks</strong> label.</li>
</ul></td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

Ecarrion committed:
[Watch] Add dependency sync support (#12683)

Ecarrion committed:
Merge trunk

Ecarrion committed:
Watch: Set dependencies as an environment value

Ecarrion committed:
remove print statement for security reasons

Ecarrion committed:
Add connect view that explains how to connect to the phone

Ecarrion committed:
Update the watch main view based on the saved credentials state

Ecarrion committed:
[Watch] Add connecting screen (#12709)

Ecarrion committed:
Merge branch 'trunk' into ec/feature-watch-app

Ecarrion committed:
Share  StoreDataService from the Widgets extension

Ecarrion committed:
Add new woo foundation framework

Ecarrion committed:
Add WooFoundationWatchOS framework. Currently only one date extension exists

Ecarrion committed:
Temporarily fetch today stats from the view

Ecarrion committed:
Reactor watch dependencies so we can accept nill dependencies as a logged out state

Ecarrion committed:
Create ViewModel and fetch stats trhough it

Ecarrion committed:
Adds documentation

Ecarrion committed:
Add CurrencySettings to WooFoundationiOS

Ecarrion committed:
Watch: Sync store name as a dependency

Ecarrion committed:
Watch: Render store name from dependencies

Ecarrion committed:
Remove unwanted whitespace in podfile

Ecarrion committed:
Update podfile

Ecarrion committed:
Merge trunk 16-04-2024

Ecarrion committed:
Merge: My Store Stats fetching data

Ecarrion committed:
Watch: Rename ContentView to MyStoreView

Ecarrion committed:
Watch: Raw ui structure for the MyStore screen

Ecarrion committed:
Update MyStoreView to hide magic numbers behind a simpler interface

Ecarrion committed:
[Watch] My Store Stats UI (#12771)

Ecarrion committed:
Merge branch 'trunk' into ec/feature-watch-app

Ecarrion committed:
Crete loading and error view

Ecarrion committed:
Shows correct data on my store stats view

Ecarrion committed:
Use correct revenue value

Ecarrion committed:
[Watch] Show correct stats data (#12774)

Ecarrion committed:
Merge branch 'trunk' into ec/feature-watch-app

Ecarrion committed:
Watch: Move store info formatting code into a shared file

Ecarrion committed:
Send store settings over

Ecarrion committed:
Use dependencies.currency settings in my store view

Ecarrion committed:
Fix swiftlint violations

Ecarrion committed:
[Watch] Share Currency Settings (#12778)

Ecarrion committed:
Merge branch 'trunk' into ec/feature-watch-app

Ecarrion committed:
[Watch] Adds empty OrderListView file

Ecarrion committed:
Watch: Add add tab view interface

Ecarrion committed:
[Watch] Add handling for changing the tab programatically

Ecarrion committed:
Main UI for navigation list card

Ecarrion committed:
Watch: Show button to go back to orders

Ecarrion committed:
Watch: Move constants to private enum

Ecarrion committed:
Watch: Make the my store tab the default one

Ecarrion committed:
[Watch] Order list UI (#12798)

Ecarrion committed:
Merge branch 'trunk' into ec/feature-watch-app

Ecarrion committed:
Watch: Add Order files to NetworkingWatchOS

Ecarrion committed:
Watch: Condiionally compile Azter HTML Parser

Ecarrion committed:
Watch: Add remanent order files

Ecarrion committed:
Watch: Copy Address copiable code

Ecarrion committed:
Watch: Add Order Data Service

Ecarrion committed:
Query: Orders list

Ecarrion committed:
Watch: Expose Order List view state via the vm

Ecarrion committed:
Watch: Show loading and error view on OrdersList screen

Ecarrion committed:
Add the necesary files to use OrderListCellViewModel on the watch

Ecarrion committed:
Watch: Properly format order list card

Ecarrion committed:
Watch: Adjust Order List Card vertical spacing

Ecarrion committed:
Add documentation

Ecarrion committed:
[Watch] Fetch and display Order List Data  (#12807)

Ecarrion committed:
Merge branch 'trunk' into ec/feature-watch-app

Ecarrion committed:
[Watch] Add empty Order Detail View

Ecarrion committed:
Watch: use navigation stack instead of NavigationSplitView

Ecarrion committed:
Watch: Add order detail UI

Ecarrion committed:
Watch: Update View.Order with more information to render

Ecarrion committed:
Watch: Render order detail in order data

Ecarrion committed:
Watch: Adjust UI for order item view

Ecarrion committed:
Watch: Render order detail into tabs

Ecarrion committed:
Watch: Handle tab selection on order details

Ecarrion committed:
Watch: Rework dividers for product list

Ecarrion committed:
Watch: Update name to spawn over two lines in the order detail view

Ecarrion committed:
Watch: Adds Order Detail View (#12817)

Ecarrion committed:
Merge branch 'trunk' into ec/feature-watch-app

Ecarrion committed:
Watch: Add the necesary files for address formatting

Ecarrion committed:
Watch: OrderDetail customer UI

Ecarrion committed:
Watch: Style ammends to the custom section in order details

Ecarrion committed:
[Watch] Add Customer Section on Order Details (#12825)

Ecarrion committed:
Merge branch 'trunk' into ec/feature-watch-app

Ecarrion committed:
Watch Copy notification file sound

Ecarrion committed:
Watch: Create app delegate and app bindings

Ecarrion committed:
Watch: Parse push notification object from user info

Ecarrion committed:
Watch: Extract code to fetch an order from a notification

Ecarrion committed:
Update PushNotification to use Int64 numbers

Ecarrion committed:
Watch: Setup logging

Ecarrion committed:
Watch: Launch order view when an order notification arrives

Ecarrion committed:
Add file to test targets

Ecarrion committed:
Watch: Add new order notification support (#12836)

Ecarrion committed:
Merge branch 'trunk' into ec/feature-watch-app

Ecarrion committed:
Watch: Add empty widget extension

mokagio committed:
Add code signing automation for Watch Extension target

mokagio committed:
Specify `us-east-2` as the S3 bucket region for Fastlane match

Fastlane warned:

> S3 client configured for "us-east-1" but the bucket
> "a8c-fastlane-match" is in "us-east-2"; Please configure the proper
> region to avoid multiple unnecessary redirects and signing attempts

Ecarrion committed:
watch: Add ApppLink complication

Ecarrion committed:
Watch: Update widgets extension identifier

Ecarrion committed:
Watch: Add widgets development profile

Ecarrion committed:
Watch: ALWAYS_EMBED_SWIFT_STANDARD_LIBRARIES to inherit to silence cocoapods warnings

Ecarrion committed:
Watch: Compile to the necessary targets

Ecarrion committed:
Add accessoryCircular to watch widgets

Ecarrion committed:
Add code signing automation for Watch App and Watch Extension targets (#12849)

mokagio committed:
Use modern code signing identities and DRY at `xcconfig` level

mokagio committed:
Remove unused `PROVISIONING_PROFILE` custom build setting

mokagio committed:
DRY development team definition to `xcconfig`

mokagio committed:
Configure provisioning profiles for Watch App extension target

mokagio committed:
Remove some redundant provisioning profiles settings

mokagio committed:
Add lanes to build App Store and Enterprise without uploading

Useful for testing build changes locally.

mokagio committed:
DRY `VERSION_LONG` usages in `Fastfile`

mokagio committed:
Fix App Center upload lane failing to read build code

Ecarrion committed:
Refactor widget so properly renders on watch

Ecarrion committed:
[Watch] Add app link complication (#12856)

Ecarrion committed:
Merge branch 'trunk' into ec/feature-watch-app

Ecarrion committed:
Add watch tracks event definitions

Ecarrion committed:
Watch: Allow watch events to be delegated to their counterpart app

Ecarrion committed:
Watch: Send watch events

Ecarrion committed:
Fix linter warning

Ecarrion committed:
Watch: Assign tracks provider to phone synchronizer

Ecarrion committed:
Watch: Add tracks events (#12895)

Ecarrion committed:
Merge branch 'trunk' into ec/feature-watch-app

Ecarrion committed:
Watch: Refactor how dependencies are synced in the phone

Ecarrion committed:
Watch: Only store new dependencies on the watch

Ecarrion committed:
Watch: make sure we always have credentials at the beggining of the sync proccess

Ecarrion committed:
Force to redraw watch app when the store changes

Ecarrion committed:
Watch: Improve the way dependencies sync (#12898)

Ecarrion committed:
Merge branch 'trunk' into ec/feature-watch-app

Ecarrion committed:
Added changelog

iangmaia commented:
**Note**: this PR [was reverted](https://github.com/woocommerce/woocommerce-ios/commit/15615447965e725c148901fd46ca6d46944ac43b) for the `18.9` release.
